springapp/src/main/java/com/javaacademy/robot/controllers/BookRestAPIController.java

```diff
@@ -27,8 +30,10 @@ public BookRestAPIController(BookService bookService, BookSearch bookSearch) {
     }
 
     @RequestMapping("/api/books")
-    public BookDto bookRequestById(@RequestParam(value = "id") Long id) {
-        return bookService.getBookByIsbn(id);
+    public ResponseEntity<BookDto> bookRequestById(@RequestParam(value = "id") Long id) {
+        BookDto foundBook = bookService.getBookByIsbn(id);
+        if(foundBook == null) return ResponseEntity.notFound().build();
```

Comment:
return Objects.nonNull(foundBook) ? 
ResponseEntity.ok(bookService.getBookByIsbn(id)) : 
ResponseEntity.notFound().build();

Reply:
this solution looks better I think